entity-framework/core/querying/complex-query-operators.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Complex Query Operators - EF Core
+author: smitpatel
+ms.date: 09/30/2019
+ms.assetid: 2e187a2a-4072-4198-9040-aaad68e424fd
+uid: core/querying/complex-query-operators
+---
+# Complex Query Operators
+
+Language Integrated Query (LINQ) contains many complex operators which combines multiple data sources or does complex processing. Not all LINQ operators have suitable translation in server side. Sometimes, query written in one form can be translated but in different form doing different computation cannot be translated. This page describes some of the complex operators and their supported variations. In future we may recognized more pattern and add their translation. Important point to remember is database also varies in what they support so a particular query which is translated in SqlServer may not work for Sqlite database.
+
+> [!TIP]
+> You can view this article's [sample](https://github.com/aspnet/EntityFramework.Docs/tree/master/samples/core/Querying) on GitHub.
+
+## Join
+
+LINQ Join operator allows you to connect two data sources based on the key selector on each source and generates a tuple of values when key matches. This naturally translates to `INNER JOIN` on relational databases. While LINQ Join has outer & inner key selectors, databases take a condition as join condition and EF Core will generate join condition by comparing outer key selector to inner key selector through equality operation. Further, if key selectors are anonymous types, EF Core will generate join condition to perform equality component wise.
```

Comment:
Would *left* and *right* be clearer than *outer* and *inner*?

Reply:
*source* might also be confusing. Maybe *set* is more idiomatic for EF users

Reply:
Open to ideas. APIs call them outer & inner source.